doc/type_requirements/File.qbk

```diff
@@ -0,0 +1,184 @@
+[/
+    Copyright (c) 2013-2016 Vinnie Falco (vinnie dot falco at gmail dot com)
+
+    Distributed under the Boost Software License, Version 1.0. (See accompanying
+    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+]
+
+[section:File File]
+
+The [*File] concept abstracts access to files in the underlying file system.
+
+To support interfaces other than Win32 or POSIX, callers may provide their
+own [*File] type that meets these requirements. The unit test code also provides
+its own [*File] type which causes simulated operating system file failures
+to exercise all failure paths in the implementation.
+
+[heading Associated Types]
+* [link nudb.ref.nudb__path_type path_type]
+* [link nudb.ref.nudb__file_mode file_mode]
+* [link nudb.ref.nudb__error_code error_code]
```

Comment:
Note that it's very common in the "valid expression" approach usually used in the documentation of concepts.